### DISCUSSION TITLE
Variant prediction example: selecting top ftrs before CV is data leakage

### DISCUSSION THREAD

**Discussion by kiramt** [facebookresearch/esm#140] [General] [answered]
Hi - I noticed in the variant prediction example notebook esm/examples/sup_variant_prediction.ipynb the top 60 principal components are selected and used to reduce the dimensionality of the training set. You're careful to separate the test set out before you do this to avoid data leakage into the test set, but by doing it before running CV there is still data leakage into the cross-validation sets. 

I imagine it doesn't affect the results much since the test set is not affected (though maybe gridsearch might have selected a different model), but probably best not to promote bad practice? 

Kira

**Comment by tomsercu** [comment, marked as answer]
Agree good call, this will throw the CV off a bit and is bad practice. Ideally we should also actually retrain with the best hyperparams on the full train split. I'd be happy to merge a PR!

**Comment by kiramt** [comment]
ok. I have the change ready but might take a while to get the CLA authorised at my end...
